### ISSUE TITLE
[display] Unhandled DisplayType when hovering enum abstract var names.

### ISSUE THREAD

**Issue by nadako** [HaxeFoundation/haxe#5166]
```haxe
@:enum abstract E(Int) {
    var A = 5;
}

class Main {
    static function main() {
    }
}
```

`haxe --display Main.hx@33@type`

```
Fatal error: exception Display.DisplayType(_, _)
Raised at file "src/typing/typer.ml", line 3797, characters 8-35
Called from file "src/typing/typecore.ml", line 239, characters 32-64
Called from file "src/typing/typeload.ml", line 2179, characters 13-74
Called from file "src/typing/typecore.ml", line 356, characters 3-6
Called from file "src/typing/type.ml", line 897, characters 13-19
Called from file "src/main.ml", line 1720, characters 34-48
Called from file "src/main.ml", line 639, characters 3-11
Called from file "src/main.ml", line 1836, characters 1-35
```

**Comment by Simn** [comment]
That's a bit scary, it comes across a TLazy when trying to print the type.

**Comment by Simn** [comment]
Logging the problem here because it might occur in other situations: The position of the field expression was set to the same position as the field itself.

The compiler determines that we want type-display for the field and correctly raises the DIsplayType exception. However, when trying to print the type it still comes across its TLazy which causes typing of the expression. Because that expression has the same position as the field, an EDisplay is inserted and consequentially causes another DisplayType exception from the typer.

Long story short we have to make sure that our positions are accurate.
